### PR TITLE
Closes #1208: Added classes to page template depending on contents of off-canvas-nav

### LINKED ISSUE #1208: Persistent main nav borders
When disabling or removing the main nav block from the navigation region (top nav) the borders remain when they shouldn't.

See above image for gray lines

### PR DESCRIPTION
**Pull request by cpoff**
## Description

When the navbar is removed the 1px grey border remained because the border CSS was assigned to the navbar-offcanvas selector, which was still in the DOM. I added logic to the twig page template to assign classes based on whether nav was populated or empty.

## Related Issue

[https://github.com/az-digital/az_quickstart/issues/1208](https://github.com/az-digital/az_quickstart/issues/1208)

## How Has This Been Tested?

I removed the navigation using several methods, most involving the Block Layout tool.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/templates/layout/page.html.twig
Comment (joeparsons) [inline themes/custom/az_barrio/templates/layout/page.html.twig:137]: I don't think Twig supports trailing commas.  Also a whitespace nitpick.
```suggestion
                    page.navigation ? 'has-navigation-region' : 'no-navigation-region',
                    page.navigation_offcanvas ? 'has-off-canvas-region' : 'no-off-canvas-region'
```

File: themes/custom/az_barrio/templates/layout/page.html.twig
Comment (joeparsons) [inline themes/custom/az_barrio/templates/layout/page.html.twig:133]: I think we still need to this `if` statement to be here.  If we don't, the `endif` on line 157 should also be removed.

File: themes/custom/az_barrio/templates/layout/page.html.twig
Comment (joeparsons) [reply themes/custom/az_barrio/templates/layout/page.html.twig:137]: Actually, the trailing comma might be fine.  I think the removal of the if statement above without removing the `endif` is what is causing the syntax error.

Comment (trackleft) [reply themes/custom/az_barrio/templates/layout/page.html.twig:137]: @joeparsons do we want to move setting of the classes to the top of the template?

**Comment by trackleft** [comment]
I think there is a file missing with the css to remove the border.

So to me the options are, add custom CSS in this PR, or add a `border-0` class instead.

https://digital.arizona.edu/arizona-bootstrap/docs/2.0/utilities/borders/#subtractive

File: themes/custom/az_barrio/templates/layout/page.html.twig
Comment (joeparsons) [reply themes/custom/az_barrio/templates/layout/page.html.twig:137]: > @joeparsons do we want to move setting of the classes to the top of the template?

Sounds reasonable to me.

**Comment by trackleft** [comment]
Should we consider this PR a bugfix, and address this:

> ☝️ also looks like there is an existing variable that is not being used `navbar_offcanvas_attributes` 🤦

In a follow-up?

**Comment by joeparsons** [comment]
> Should we consider this PR a bugfix, and address this:
> 
> > ☝️ also looks like there is an existing variable that is not being used `navbar_offcanvas_attributes` 🤦
> 
> In a follow-up? Since there are Google Tag Manager implications?

I think that sounds reasonable.  What are the GTM implications?

**Comment by trackleft** [comment]
> I think that sounds reasonable. What are the GTM implications?

Changing the ID could cause some tracking code to cease working.

File: themes/custom/az_barrio/templates/layout/page.html.twig
Comment (cpoff) [reply themes/custom/az_barrio/templates/layout/page.html.twig]: Ok. That snippet now moved up below the comments and before the opening page-wrapper.

File: themes/custom/az_barrio/css/az-barrio-off-canvas-nav.css
Comment (trackleft) [inline themes/custom/az_barrio/css/az-barrio-off-canvas-nav.css:8]:
```suggestion
}

```

Comment (cpoff) [reply themes/custom/az_barrio/css/az-barrio-off-canvas-nav.css:8]: Done

**Review by trackleft — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]
Changes look good to me now.  We should be sure to create a ticket for the follow-up tasks before merging.

**Comment by danahertzberg** [comment]
Works with block being disabled, but not if menu links are all disabled (didn't test if menu links are deleted). This is easy enough to communicate and should not block merging.

**Review by danahertzberg — APPROVED** [review]

**Pull request merged**
